modules/openid_auth/app/controllers/openid_auth/v0/okta_controller.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'json'
+
+require_dependency 'openid_auth/application_controller'
+
+module OpenidAuth
+  module V0
+    class OktaController < ApplicationController
+      skip_before_action :authenticate
+
+      def metadata
+        meta = OneLogin::RubySaml::Metadata.new
+        render xml: meta.generate(saml_settings), content_type: 'application/xml'
+      end
+
+      def parse_user_attributes(json_request)
+        okta_attributes = json_request.dig(:data, :assertion, :claims)
+        user_attributes = Hash.new
+        okta_attributes.each do |k, v|
+          user_attributes[k.to_sym] = v[:attributeValues][0][:value]
+        end
+        user_attributes
+      end
+
+      def fetch_mvi_profile(user_attributes)    
+        user_identity = OpenidUserIdentity.new(
```

Comment:
@patrickvinograd per your suggestion, creating the new user identity directly. I think the icn is the only thing technically required to get what we need in the current flow, but not sure what other attributes we want to include, if any